### DISCUSSION TITLE
Is there a way to set a timeout or an attempt count for negotiations?

### DISCUSSION THREAD

**Discussion by dcandas** [eclipse-edc/Connector#1362] [Q&A]
Our controller has to handle a lot of negotiations and it becomes unreachable if too many negotiations are stuck in requested. I think the requests are constantly repeated. Is there a way to set a timeout or an attempt count for negotiations?

**Comment by ndr-brt** [comment]
Can you describe this scenario in more details?

**Reply by dcandas** [reply]
About 250 negotiations and then immediately corresponding transactions are started by multiple connectors. Some of these negotiations and transactions fail due to misconfigured webhooks or unreachable servers. Our controller repeats a lot of negotiations or transactions which we would rather fail after a set number of attempts instead of causing the controller to become unreachable.

**Reply by ndr-brt** [reply]
Can you list the steps needed to reproduce the issue and report logs/stack traces? 
I just tried to negotiate 250 contracts between two connectors locally and they went all confirmed in few seconds

**Reply by dcandas** [reply]
The problem is when one of the requesting connectors are somehow misconfigured. Negotiations which should eventually fail are stuck on repeat. Is there a way to set a timeout or attempt count for this scenario?
